git version older then 1.8.5 unsupport -C option

<!-- 

Please try to use English to describe your issue, or at least provide a snippet of English translation.

-->

## Issue Description

Type: *bug report* or *feature request*

Error encountered executing make command，the reason is git git version older then 1.8.5 unsupport -C option

git -C target/cache/chaosblade-exec-os pull origin v0.4.0
Unknown option: -C
usage: git [--version] [--help] [-c name=value]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]
make: *** [build_os] Error 129

### Describe what happened (or what feature you want)


### Describe what you expected to happen

Give friendly tips to tell user how to resolve it

### How to reproduce it (as minimally and precisely as possible)

1. 
2. 
3. 

### Tell us your environment


### Anything else we need to know?

--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,13 @@
 
 export BLADE_VERSION=0.4.0
 
+ALLOWGITVERSION=1.8.5
+GITVERSION:=$(shell git --version | grep ^git | sed 's/^.* //g')
+
+ifneq ($(strip $(firstword $(sort $(GITVERSION), $(ALLOWGITVERSION)))),$(ALLOWGITVERSION))
+	ALERTMSG="please update git to >= $(ALLOWGITVERSION)"
+endif
+
 BLADE_BIN=blade
 BLADE_EXPORT=chaosblade-$(BLADE_VERSION).tgz
 BLADE_SRC_ROOT=`pwd`
@@ -123,6 +130,9 @@ endif
 
 # create dir or download necessary file
 pre_build:mkdir_build_target download_sandbox download_blade_java_agent download_cplus_agent
+	ifdef ALERTMSG
+		$(error $(ALERTMSG))
+	endif
 	rm -rf $(BUILD_TARGET_PKG_DIR) $(BUILD_TARGET_PKG_FILE_PATH)
 	mkdir -p $(BUILD_TARGET_BIN) $(BUILD_TARGET_LIB)
 	# unzip jvm-sandbox

--- a/Makefile
+++ b/Makefile
@@ -105,6 +105,9 @@ build_os:
 ifneq ($(BUILD_TARGET_CACHE)/chaosblade-exec-os, $(wildcard $(BUILD_TARGET_CACHE)/chaosblade-exec-os))
 	git clone -b $(BLADE_EXEC_OS_BRANCH) $(BLADE_EXEC_OS_PROJECT) $(BUILD_TARGET_CACHE)/chaosblade-exec-os
 else
+	ifdef ALERTMSG
+		$(error $(ALERTMSG))
+	endif
 	git -C $(BUILD_TARGET_CACHE)/chaosblade-exec-os pull origin $(BLADE_EXEC_OS_BRANCH)
 endif
 	make -C $(BUILD_TARGET_CACHE)/chaosblade-exec-os
@@ -130,9 +133,6 @@ endif
 
 # create dir or download necessary file
 pre_build:mkdir_build_target download_sandbox download_blade_java_agent download_cplus_agent
-	ifdef ALERTMSG
-		$(error $(ALERTMSG))
-	endif
 	rm -rf $(BUILD_TARGET_PKG_DIR) $(BUILD_TARGET_PKG_FILE_PATH)
 	mkdir -p $(BUILD_TARGET_BIN) $(BUILD_TARGET_LIB)
 	# unzip jvm-sandbox

